File: pkg/galaxy/division.go
package galaxy

type Division struct {
	ID              string `json:"id"`
	ResourcesAmount int    `json:"resources_amount"`
	TechAttack      int    `json:"tech_attack"`
	TechDefense     int    `json:"tech_defense"`
	TechEngines     int    `json:"tech_engines"`
	TechCargo       int    `json:"tech_cargo"`
}

File: assets/js/entities/division.js
export class Division {
    /** @type {string} */
    id = '';

    /** @type {number} */
    resources_amount = 0;

    /** @type {number} */
    tech_attack = 0;

    /** @type {number} */
    tech_defense = 0;

    /** @type {number} */
    tech_engines = 0;

    /** @type {number} */
    tech_cargo = 0;

    /**
     * @param {Object} data
     * @returns {Division}
     */
    updateFromDTO(data) {
        this.id = data.id;
        this.resources_amount = data.resources_amount ?? 0;
        this.tech_attack = data.tech_attack ?? 0;
        this.tech_defense = data.tech_defense ?? 0;
        this.tech_engines = data.tech_engines ?? 0;
        this.tech_cargo = data.tech_cargo ?? 0;
        return this;
    }
}
